### PR TITLE
[#809] Update HTML report section in DG

### LINKED ISSUE #809: Outdated HTML report section in DG
The current HTML Report section in our DG
- contains an outdated report UI.
- should be `index.pug` instead of `index.jade` 

Screen Shot 2019-07-12 at 11 29 29 AM

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes #809 

```
The current HTML report section in our DG contains outdated info such as
the report summary diagram, sequence diagrams and the list of js files.

Let's update the HTML report section with the latest report summary
diagram, Vue lifecycle diagram, sequence diagrams and the descriptions
of the recently added js files such as v_ramp, v_zoom, and v_segment.
```

### CODE REVIEW

**Comment by jamessspanggg** [comment]
Updated DG section:

Screen Shot 2019-07-12 at 12 11 05 PM

**Comment by damithc** [comment]
Instead of curly braces, use a line to connect the description to the box?

**Comment by jamessspanggg** [comment]
> Instead of curly braces, use a line to connect the description to the box?

ok noted.

**Comment by fzdy1914** [comment]
> Instead of curly braces, use a line to connect the description to the box?

Prof, do you think an arrow is more suitable than a line?

**Comment by damithc** [comment]
> Prof, do you think an arrow is more suitable than a line?

That can work too.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -316,13 +319,22 @@ The summary module is activated after the information is loaded from the main Vu
...
 The authorship module retrieves the relevant information from the corresponding `authorship.json` file if it is not yet loaded. If it has been loaded, the data will be written into `window.REPOS` and be read from there instead.
 
 ![authorship architecture](images/report-architecture-authorship.png)
 
 #### Showing relevant information by authors
-The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code to be displayed in the tab view which will be popped up on the right side of the screen.
+The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code segments to be displayed in the tab view which will be popped up on the right side of the screen.
+
+### Zoom View (v_zoom.js)
+The `v_zoom` module is in charge of filtering and displaying the sub-range of commits from the selected ramp chart.
```
Comment (eugenepeh) [inline docs/DeveloperGuide.md]: the commits from selected sub-range of a ramp chart would be better.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -316,13 +319,22 @@ The summary module is activated after the information is loaded from the main Vu
...
 
 #### Showing relevant information by authors
-The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code to be displayed in the tab view which will be popped up on the right side of the screen.
+The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code segments to be displayed in the tab view which will be popped up on the right side of the screen.
+
+### Zoom View (v_zoom.js)
+The `v_zoom` module is in charge of filtering and displaying the sub-range of commits from the selected ramp chart. 
+
+### Ramp View (v_ramp.js)
+The `v_ramp` module is responsible to retrieve the relevant information from `v_summary` and generate ramp charts that contain ramp slices.
```
Comment (eugenepeh) [inline docs/DeveloperGuide.md]: why is it retrieve and not receive? When the dependency is one way, v_ramp should have no knowledge of its parent.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -261,13 +261,16 @@ The main Vue object (`window.app`) is responsible for the loading of the report
...
 ### Javascript Files
 - [**main.js**](../frontend/src/static/js/main.js) - main controller that pushes content into different modules
 - [**api.js**](../frontend/src/static/js/api.js)- loading and parsing of the report content
-- [**v_summary.js**](../frontend/src/static/js/v_summary.js) - module that supports the ramp chart view
-- [**v_authorship.js**](../frontend/src/static/js/v_authorship.js) - module that supports the authorship view
+- [**v_summary.js**](../frontend/src/static/js/v_summary.js) - module that supports the summary view
+- [**v_authorship.js**](../frontend/src/static/js/v_authorship.js) - module that supports the authorship tab view
+- [**v_zoom.js**](../frontend/src/static/js/v_zoom.js) - module that supports the zoom tab view
+- [**v_ramp.js**](../frontend/src/static/js/v_ramp.js) - module that supports the ramp chart view
+- [**v_segment.js**](../frontend/src/static/js/v_segment.js) - module that supports the code segment view
```
Comment (eugenepeh) [inline docs/DeveloperGuide.md]: A little change here, can I have these jump to the each sub sections and the subsection header e.g. The (main.js) in ### Main (main.js) directs user to the file?

**Comment by eugenepeh** [comment]
@reposense/devs please help to take a look if there's anything requires supplement

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -316,13 +319,22 @@ The summary module is activated after the information is loaded from the main Vu
...
 
 #### Showing relevant information by authors
-The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code to be displayed in the tab view which will be popped up on the right side of the screen.
+The files will be filtered, picking only files the selected author has written in. The lines are then split into chunks of "touched" and "untouched" code segments to be displayed in the tab view which will be popped up on the right side of the screen.
+
+### Zoom View ([v_zoom.js](../frontend/src/static/js/v_zoom.js))
+The `v_zoom` module is in charge of filtering and displaying the commits from selected sub-range of a ramp chart. 
+
+### Ramp View ([v_ramp.js](../frontend/src/static/js/v_ramp.js))
+The `v_ramp` module is responsible to receive the relevant information from `v_summary` and generate ramp charts that contain ramp slices.
```
Comment (bluein-green) [inline docs/DeveloperGuide.md]: responsible **for receiving** the relevant information .... and **generating** ramp charts

**Comment by jamessspanggg** [comment]
Added The Vue lifecycle diagram as it is more suitable to be under the HTML Report section.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -316,13 +321,22 @@ The summary module is activated after the information is loaded from the main Vu
...
+
+#### Padding for dates
+For ramps between the date ranges, the slices will be selected and it will be pre and post padded with empty slices to align the ramp slice between the `sinceDate` and `untilDate`. The ramps will then be rendered with the slices in the right position.
+
+### Segment View ([v_segment.js](../frontend/src/static/js/v_segment.js))
+The `v-segment` module separates the code in terms of "touched" and "untouched" segments and only loads each "untouched" segment when it is toggled.
```
Comment (bluein-green) [inline docs/DeveloperGuide.md]: You could specify here that `v_segment` is used as a component in `v_authorship`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -245,10 +245,12 @@ gradlew run -Dargs="-c ./configs/ -o output_path/ -s 21/10/2017 -u 21/11/2017 -f
...
 
-The main HTML file is generated from [`frontend/src/index.jade`](../frontend/src/index.jade).
+The main HTML file is generated from [`frontend/src/index.pug`](../frontend/src/index.pug).
 
-[Vue](https://vuejs.org/v2/api/) (pronounced /vjuː/, like view) is a progressive framework for building user interfaces. It is heavily utilized in the report to dynamically update the information in the various views. (Style guide available [here](https://vuejs.org/v2/style-guide/), Developer tool available [here](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd)).
+[Vue](https://vuejs.org/v2/api/) (pronounced /vjuː/, like view) is a progressive framework for building user interfaces. It is heavily utilized in the report to dynamically update the information in the various views. (Style guide available [here](https://vuejs.org/v2/style-guide/), Developer tool available [here](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd)). Vue lifecycle hooks are the defined methods which gets executed in a certain stage of the Vue object lifespan. The following is the lifecycle diagram indicating the hook sequence:
```
Comment (bluein-green) [inline docs/DeveloperGuide.md]: It would be good to specify the source of the diagram, since we didn't produce it

Comment (jylee-git) [inline docs/DeveloperGuide.md]: In docs/images/vue-lifecycle-diagram.png :

Might not be a good idea to split into two halves. Either draw a dotted line to link the arrows together, paste the image from https://vuejs.org/images/lifecycle.png as a whole, or link to [this page](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram) instead for them to read up on.

Also adding on to @bluein-green , if pasting the image file, do ensure that they allow modification and/or reproduction of the image file.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -316,13 +321,22 @@ The summary module is activated after the information is loaded from the main Vu
...
+The `v_zoom` module is in charge of filtering and displaying the commits from selected sub-range of a ramp chart. 
+
+### Ramp View ([v_ramp.js](../frontend/src/static/js/v_ramp.js))
+The `v_ramp` module is responsible for receiving the relevant information from `v_summary` and generating ramp charts that contain ramp slices.
+
+#### Padding for dates
+For ramps between the date ranges, the slices will be selected and it will be pre and post padded with empty slices to align the ramp slice between the `sinceDate` and `untilDate`. The ramps will then be rendered with the slices in the right position.
+
+### Segment View ([v_segment.js](../frontend/src/static/js/v_segment.js))
+The `v-segment` module is used as a component in `v_authorship`. It separates the code in terms of "touched" and "untouched" segments and only loads each "untouched" segment when it is toggled.
```
Comment (jylee-git) [inline docs/DeveloperGuide.md]: ### Suggestion:

... (The module) distinguishes code segments that are touched by an author and those that are not. (The module) also handles the loading of untouched code segments when the user clicks on the `➕` icon.

**Comment by jamessspanggg** [comment]
@eugenepeh ready to merge?

**Pull request merged**